NETOBSERV-316 Console plugin as standalone

jpinsonneau opened the pull request:
This allow to run plugin as standalone. You can `port-forward` loki from anywhere or mock it.

Simply run `make start-standalone` to run backend + standalone frontend with live reloading
Or `build-standalone` then `make serve`.

Generated `index.html` + js files are hosted in backend at http://localhost:9001/

![image](https://user-images.githubusercontent.com/91894519/169827693-057fae54-4c8c-417c-a85d-ddd92fe64285.png)
![image](https://user-images.githubusercontent.com/91894519/169827734-675b37a9-adf3-4eb3-aff4-d54472461bc2.png)

You can run [cypress testing](https://github.com/netobserv/network-observability-console-plugin/pull/155) using this method. Simply update port number in [consts.js](https://github.com/netobserv/network-observability-console-plugin/blob/main/web/cypress/support/const.js#L2)

Will need https://github.com/netobserv/network-observability-console-plugin/pull/152 to support dark theme & implements `ResourceLink` icons. This can be added in another PR.

OlivierCazade commented on Makefile:
```diff
@@ -127,6 +135,12 @@ start: build-backend
 	bash -c "trap 'fuser -k 9002/tcp' EXIT; \
 					./plugin-backend -port 9002 & cd web && npm run start" 
 
+.PHONY: start-standalone
```
May be start-standalone should also depends on install-frontend otherwise, dependency might be missing.

OlivierCazade reviewed (approved):
/lgtm

I just had to run install-frontend before start-standalone to make it work.
Thanks!

jpinsonneau commented:
> I just had to run install-frontend before start-standalone to make it work. Thanks!

Yes I can add this on both `start` / `start-standalone` :+1: 
Thanks for the feedback @OlivierCazade

OlivierCazade commented:
/lgtm

eranra commented:
@jpinsonneau just ran that on my machine ... looks good to me. some remarks::

(1) the term `install` in the Makefile is a bit confusing ... it might be better to use something like `deploy` or/and `build` those terms are easier to understand and not ambiguous.

(2) For completeness, it might be nice to have some mock Loki (or simulation) of data so that when you run the plug-in without real data we will be able to show something. 

(3) From looking on the screenshot - need to add the nice icons back :-)

jpinsonneau commented:
> (1) the term `install` in the Makefile is a bit confusing ... it might be better to use something like `deploy` or/and `build` those terms are easier to understand and not ambiguous.

This actually come from `npm install`. We can add `dependencies` keyword as same as in the echo command or maybe `npm` mention ? 
```
.PHONY: install-frontend
install-frontend:
	@echo "### Installing frontend dependencies"
	cd web && npm ${NPM_INSTALL}
```
Or remove these and run `npm install` directly. What would you prefer ?

> (2) For completeness, it might be nice to have some mock Loki (or simulation) of data so that when you run the plug-in without real data we will be able to show something.

I'm on it :+1: currently we have the [json-server](https://github.com/netobserv/documents/blob/main/hack_loki.md#fake-loki-frontend) but the jsons are outdated. I'm also looking if we can have something better like reusing mocks from tests or restore a loki storage from previous saved data.

> (3) From looking on the screenshot - need to add the nice icons back :-)
This will be done after https://github.com/netobserv/network-observability-console-plugin/pull/152 merge 

Thanks for the feedback @eranra !

jpinsonneau commented:
Rebased and added missing resource icons:

![image](https://user-images.githubusercontent.com/91894519/171375434-9072da00-69d2-41ce-90dc-911c63f36e1c.png)

It seems there are issue with button styles in dark :thinking: The PF colors are not dynamically updating on theme change and I can't find a workaround for now.

jpinsonneau commented:
Rebased to test with `cypress` and fixed lasts styling issues
Also added `start-standalone-mock` to run standalone with [mocked loki backend](https://github.com/netobserv/network-observability-console-plugin/pull/167)

![image](https://user-images.githubusercontent.com/91894519/171627563-6e2ac315-4c11-4d5e-843c-c2b0eb2ccbee.png)
![image](https://user-images.githubusercontent.com/91894519/171627627-3ce28a07-b6ee-4da8-810e-f80a8623982e.png)
![image](https://user-images.githubusercontent.com/91894519/171627678-8779e990-798c-4919-abcf-6ac24ecb9822.png)
![image](https://user-images.githubusercontent.com/91894519/171627741-789b9a0e-788d-4f44-9ba0-0baeed2c6eb9.png)
![image](https://user-images.githubusercontent.com/91894519/171627800-3d316450-f6fe-49e1-9b44-d92d71c2ed04.png)
